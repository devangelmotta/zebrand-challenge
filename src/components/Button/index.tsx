import React, { memo } from 'react';
import styled from 'styled-components';

interface Props {
  readonly display? : string,
  readonly flexDirection? : string,
  readonly justifyContent? : string,
  readonly position? : string,
  readonly alignContent? : string,
  readonly alignItems? : string,
  readonly width? : string,
  readonly height? : string,
  readonly maxHeight? : string,
  readonly minWidth? : string,
  readonly maxWidth? : string,
  readonly backgroundColor? : string,
  readonly padding? : string,
  readonly border? : string,
  readonly borderRadius? : string,
  readonly margin? : string,
  readonly hoverBgColor? : string,
  readonly hoverBorder? : string,
  readonly zIndex?: string
}

// eslint-disable-next-line @typescript-eslint/no-unused-vars
export const Button = styled.button`
  display: ${(props: Props) => props.display || "flex"};
  display: ${(props: Props) => props.position || "relative"};
  flex-direction: ${props => props.flexDirection || "row"};
  justify-content: ${props => props.justifyContent || "space-between"};
  align-content: ${props => props.alignContent || "center"};
  align-items: ${props => props.alignItems || "center"};
  width: ${props => props.width || "100%"};
  height: ${props => props.height || "auto"};
  background-color: ${props => props.backgroundColor || "transparent"};
  border: ${props => props.border || "1px solid #fff"};
  border-radius: ${props => props.borderRadius || "0"};
  padding: ${props => props.padding || "0 0 0 0"};
  margin: ${props => props.margin || "0 0 0 0"};
  z-index: ${props => props.zIndex || "auto"};
    &:hover{
      //
    }
  `;

export default memo(Button);

