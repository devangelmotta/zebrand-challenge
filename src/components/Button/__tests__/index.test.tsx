import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components'
import Button from '..';

test('Check button css', () => {
  const treeButton = renderer.create(<Button />).toJSON()
  // expect(treeButton).toMatchSnapshot()
  // expect(treeButton).toHaveStyleRule('display', 'flex')
  // expect(treeButton).toHaveStyleRule('position', 'relative')
  // expect(treeButton).toHaveStyleRule('flex-direction', 'row')
  // expect(treeButton).toHaveStyleRule('justify-content', 'center')
  // expect(treeButton).toHaveStyleRule('align-content', 'center')
  // expect(treeButton).toHaveStyleRule('align-items', 'center')
  // expect(treeButton).toHaveStyleRule('width', '100%')
  // expect(treeButton).toHaveStyleRule('height', 'auto')
  // expect(treeButton).toHaveStyleRule('background-color', 'transparent')
  // expect(treeButton).toHaveStyleRule('padding', '0 0 0 0')
  // expect(treeButton).toHaveStyleRule('border', '0 solid transparent')
  // expect(treeButton).toHaveStyleRule('border-radius', '0')
  // expect(treeButton).toHaveStyleRule('margin', '0 0 0 0')
  // expect(treeButton).toHaveStyleRule('z-index', 'auto')
})
