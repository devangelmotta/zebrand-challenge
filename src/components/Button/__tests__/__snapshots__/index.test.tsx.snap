// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check button css 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  display: relative;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: auto;
  height: auto;
  background-color: transparent;
  border: 1px solid #999;
  border-radius: 0;
  padding: 0 0 0 0;
  margin: 0 0 0 0;
  z-index: auto;
}

<button
  className="c0"
/>
`;
